docs/data/data-grid/editing/editing.md

```diff
@@ -4,266 +4,411 @@ title: Data Grid - Editing
+{{"demo": "AutoStopEditComponent.js", "bg": "inline", "defaultCodeOpen": false}}
 
-### Control with external buttons [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+> ⚠ We don't recommend using edit components with auto-stop in columns that use long-running `preProcessEditCellProps` because the UI will freeze while waiting for `apiRef.current.setEditCellValue`.
+> Instead, use the provided stop editing interactions to stop the edit mode.
 
-You can [disable the default behavior](/components/data-grid/events/#disabling-the-default-behavior) of the grid and control the row edit using external buttons.
+## Row editing
 
-Here is shown how a full-featured CRUD can be created.
+Row editing allows to edit all the cells of a row at once.
+It is based on the cell editing, so everything written above remains valid.
+To enable it, change the `editMode` prop to `"row"`, then set normally the `editable` property in the definition of the columns that should be editable.
 
-{{"demo": "FullFeaturedCrudGrid.js", "bg": "inline", "disableAd": true}}
+```tsx
+<DataGrid editMode="row" columns={[{ field: 'name', editable: true }]} />
+```
 
-### Saving rows with nested structures
+Below there is a demo with row editing.
+To [start](#start-editing) and [stop](#stop-editing) editing use the same shortcuts from the cell editing (e.g. double-clicking a cell).
 
-Saving columns that make use of `valueGetter` can be done adding a `valueSetter`.
-The same [approach](/components/data-grid/editing/#saving-nested-structures) from the cell editing mode can be used here.
-Note that the `valueSetter` will be called for each field.
+{{"demo": "BasicRowEditingGrid.js", "bg": "inline", "defaultCodeOpen": false}}
 
-### Server-side persistence [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+> ⚠ By design, when changing the value of a cell all `preProcessEditCellProps` callbacks from other columns are also called.
+> This happens to allow conditional validation (explained below) where the value of a cell impacts on the validation status of a another cell in the same row.
+> To only run validation if the value has changed check if the `hasChanged` param is `true`.
 
-If you want to send the updated data to your server, you can use the `onRowEditCommit` which is fired just before committing the new cell value to the grid.
+### Conditional validation
 
-To access the new values for the row, use `apiRef.current.getEditRowsModel` to get all rows in edit mode, then use the id provided to get only the values for the row that was committed.
+Having all cells of a row in edit mode allows validating a field based on the value of another one.
+To do that, start by adding the `preProcessEditCellProps` as explained in the [validation](#validation) section.
+When the callback is called, it will have an additional `otherFieldsProps` param contaning the props from the other fields in the same row.
```

Comment:
```suggestion
When the callback is called, it will have an additional `otherFieldsProps` param containing the props from the other fields in the same row.
```